### ISSUE TITLE
Feature: `policy_is_permissive` to check permissive/restrictive

### ISSUE THREAD

**Issue by iamed2** [theory/pgtap#343]
This would sit parallel to `policy_cmd_is`, checking `pg_catalog.pg_policy.polpermissive` (boolean not null) (introduced in PostgreSQL 10).

Implementation of the base case:

```sql
-- policy_is_permissive( schema, table, policy, description )
CREATE OR REPLACE FUNCTION policy_is_permissive( NAME, NAME, NAME, text )
RETURNS TEXT AS $$
DECLARE
    permissive boolean;
BEGIN
    SELECT pp.polpermissive
      FROM pg_catalog.pg_policy AS pp
      JOIN pg_catalog.pg_class AS pc ON pc.oid = pp.polrelid
      JOIN pg_catalog.pg_namespace AS pn ON pn.oid = pc.relnamespace
     WHERE pn.nspname = $1
       AND pc.relname = $2
       AND pp.polname = $3
      INTO permissive;

    RETURN ok( permissive, $4 );
END;
$$ LANGUAGE plpgsql;
```

and then either `policy_is_restrictive` or `policy_isnt_permissive`.

**Comment by theory** [comment]
Should there also be `policy_is_restrictive()`? Or one function to cover both?

**Comment by iamed2** [comment]
I think it makes sense to have that, yes. Having two functions goes along with other e.g. `is`/`isnt` conditions, and I now think `policy_is_restrictive` is more intuitive than `policy_isnt_permissive` (which matches how PostgreSQL stores the flag, but not how people write policies).

**Comment by theory** [comment]
Agreed that `policy_is_restrictive` is a better name (it's generally better to use positives in function names rather than negatives).

**Comment by iamed2** [comment]
I'll set some time for it on Thursday and see what I can accomplish.

I think I understand everything I need to do except how the [compat](https://github.com/theory/pgtap/tree/main/compat) stuff works (this is a 10+ feature). Is it enough to create one patch file like [this one](https://github.com/theory/pgtap/blob/2e4be23bb5b704f8a8b2bcc097680a9cb89e29c8/compat/9.4/pgtap--0.99.0--1.0.0.patch) for 9.6 for my migration SQL, or is there more? Would my tests need to indicate somehow that they're 10+?
